### PR TITLE
Disables Wield Delay on Everything except Heavy Weapons

### PR DESCRIPTION
**Pull request by ValoTheValo**
## About The Pull Request
everyone I have talked to disliked the wield delay PR in some fashion. i removed the delay on all weapons that aren't LMGs, RPGs and AMRs. so you can go back to your normal Eris experience.

More Detailed Explanation as to my thought process

Like if i get this correct from reading PR and SPCRs words these are the reasons why wield delay was added

1:Ambush combat is bad

2:Vig Needs more things to do

3: Balance OP Guns

lets start with point 1. "this makes ambush combat less effective" this change in effect is a double edged sword as it makes ambushes a lot more deadly when they occur as you cannot react in time by pulling our your weapon and returning fire as you are dead by the time it happens. it slightly lessens pulling out gun and mag-dumping at point blank range but if you have high enough VIG it is completely ignored and is buffed, as your opponent wont have enough time to pull out thier weapon and wield it in time.

"Vig needs more things to do"
VIG does need more noticeable effects but taking a thing that was normally for everything and slapping it behind a VIG requirement isn't helping anyone except powergamers. they are plenty of less annoying and better balanced ways to add VIG effects into the game that can be thought up

"Balance OP Guns"
this doesn't do much to curtail "OP guns" as instead of balancing the weapons directly it takes an extremely roundabout solution and doesn't really change anything about the weapon or its use for power-gamers. hell this PR buffed the most powerful gun in the game. the Dallas by giving it more AP. in the end it just becomes another factor in having to balance guns rather than making the game more balanced in a less interruptive way. you could easily just tweak damage, AP, fire rate, rarity etc. to get a better result. a Vintorez is just as deadly as before due to it and everything else being put on delay.

In summary Wield Delay runs counter to the things it was meant to fix or does not improve the situation at all and only serves to worsen gameplay experience and annoy players

## Why It's Good For The Game
More Fun and less sluggish combat. majority of people dislike gun wield time

## Changelog
:cl:
balance: Removed gun wield delay on all Pistols, Shotguns, Submachineguns, most rifles and most energy guns
balance: Edited delay and vig values on LMG and RPG 
/:cl:

### CODE REVIEW

File: code/modules/projectiles/guns/projectile/automatic/atreides.dm [hunk 1]
```diff
@@ -21,8 +21,8 @@
 	recoil_buildup = 1.2
 	one_hand_penalty = 5 //smg level
 	gun_tags = list(GUN_SILENCABLE, GUN_GILDABLE)
-	wield_delay = 0.5 SECOND
-	wield_delay_factor = 0.1 // 10 vig for instant
+	wield_delay = 0 SECOND
+	wield_delay_factor = 0
```
Comment (Humonitarian) [inline code/modules/projectiles/guns/projectile/automatic/atreides.dm:25]: Inherited from code/modules/projectiles/guns/projectile/automatic.dm, can be removed

Comment (Humonitarian) [reply code/modules/projectiles/guns/projectile/automatic/atreides.dm:25]: Same goes for all the rest too

**Comment by MLGTASTICa** [comment]
Litteraly a revert PR that is a cope reaction to now needing vigilance to game on people with heavy guns.

**Comment by Kegdo** [comment]
Honestly, I like the wield timer. Why? Because there is no wield timer for melee weapons. Ergo, I can game on heavy weapon users
Also __just use the fucking melee__ if you don't like the VIG stat scaling~~, it's that simple~~

**Comment by Humonitarian** [comment]
> Honestly, I like the wield timer. Why? Because there is no wield timer for melee weapons. Ergo, I can game on heavy weapon users Also **just use the fucking melee** if you don't like the VIG stat scaling, ~it's that simple~

Imagine a gang of scientists armed with swords running at a single operative armed with a wintermute. Only fun for one person 🤣

**Comment by ValoTheValo** [comment]
> Litteraly a revert PR that is a cope reaction to now needing vigilance to game on people with heavy guns.

You still need vig for heavy guns.

**Comment by MLGTASTICa** [comment]
This PR overall does nothing to help the balance of the game in the right direction , it directly adds wield delay to the AMR and Valkrie (both are guns that are very limited in shots and SHOULDNT have a delay for their limit) , ontop of this , the PR further reduces the wield delays on LMGS by another 0.5 seconds and removes them all togheter from the heavy assault rifles (Wintermule and Vintorez). This PR serves no scope in the interest of actual balance and instead is single-handedly done to keep LMGS worse-overall in CQC by removing the application of vigilance in wielding in general, and only limiting it to LMGs.
The 3 arguments specified are also very weak , this is not intended to adress proper ambushes , but instant CQC wielding , examples where Guild would sit next to IH , instantly wield a rifle and fire on full auto , instantly killing them. Proper ambushes can't and will never be adressed by gun delay
For the second effect , this very much makes sense to be something influenced by vigilance and IT is a mechanic. If you think you can do "better" you are welcome to code and PR a new feature all togheter instead of doing a flimsy disable and never coding in a replacement . For maintainers , i very very much don't recommend merging this PR unless a PR to add a replacement mechanic to make VIG count more is added.
The wield delay very much does a lot to curtail OP weapons. Also ,having stats is NOT powergaming. If you want vigilance round-start than pick one of the actual round-start backgrounds that give you vigilance or jobs, instead of complaining about your wield timers when you pick round-start backgrounds that decrease your vigilance to the minus, causing it to add more onto the delays.
Ontop of this , weapon balance can in the current state be done only through Damage , AP , Magsize , Size , Cost and rarity , Firerate cannot be modified due to the way code is handled and the VERY real effects of it being able to cause lag in the right conditions, adding more mechanics to balance weapons that are influencable by stats and make sense was very much needed.

**Comment by Humonitarian** [comment]
> it directly adds wield delay to the AMR and Valkrie

Good catch, these weapons should have no delay.

> is single-handedly done to keep LMGS worse-overall in CQC

Good. LMGs should be hard to deploy, but powerful. That's interesting. Being just bigger mag rifles isn't.

> to make VIG count more is added

Vig is already essential for headshots, and provides many other combat advantages.

> weapon balance can in the current state be done only through Damage , AP , Magsize , Size , Cost and rarity 

Recoil, handedness, etc. are also important. More attention should be given to those qualities.
Wield delay is good, but only when it doesn't affect weapons like pistols, or even SMGs. I wouldn't mind it on battlerifles, however, if Valo can be convinced.

As a final note, I still stand by my opinion of having a  hard-limit to how much vigilance can effect it. Instant wield on LMGs is just... Yeah, even when you have 100 vig.

**Comment by ValoTheValo** [comment]
> This PR overall does nothing to help the balance of the game in the right direction , it directly adds wield delay to the AMR and Valkrie (both are guns that are very limited in shots and SHOULDNT have a delay for their limit) , ontop of this , the PR further reduces the wield delays on LMGS by another 0.5 seconds and removes them all togheter from the heavy assault rifles (Wintermule and Vintorez). This PR serves no scope in the interest of actual balance and instead is single-handedly done to keep LMGS worse-overall in CQC by removing the application of vigilance in wielding in general, and only limiting it to LMGs. The 3 arguments specified are also very weak , this is not intended to adress proper ambushes , but instant CQC wielding , examples where Guild would sit next to IH , instantly wield a rifle and fire on full auto , instantly killing them. Proper ambushes can't and will never be adressed by gun delay For the second effect , this very much makes sense to be something influenced by vigilance and IT is a mechanic. If you think you can do "better" you are welcome to code and PR a new feature all togheter instead of doing a flimsy disable and never coding in a replacement . For maintainers , i very very much don't recommend merging this PR unless a PR to add a replacement mechanic to make VIG count more is added. The wield delay very much does a lot to curtail OP weapons. Also ,having stats is NOT powergaming. If you want vigilance round-start than pick one of the actual round-start backgrounds that give you vigilance or jobs, instead of complaining about your wield timers when you pick round-start backgrounds that decrease your vigilance to the minus, causing it to add more onto the delays. Ontop of this , weapon balance can in the current state be done only through Damage , AP , Magsize , Size , Cost and rarity , Firerate cannot be modified due to the way code is handled and the VERY real effects of it being able to cause lag in the right conditions, adding more mechanics to balance weapons that are influencable by stats and make sense was very much needed.

The anti matriel rifle is un changed. Only the valkyrie. I tested it on it and it wasnt fun to use. 

As for balance of vintorez and new vig features i dont want to bundle mostly unrelated things in same PR. And the delay on lmgs being slightly reduced was to account for the fact most other guns now look like before but if you want me to lower it more i can.

File: code/modules/projectiles/guns/projectile/automatic/atreides.dm (hunk 1 above)
Comment (ValoTheValo) [reply code/modules/projectiles/guns/projectile/automatic/atreides.dm]: Just removed it. Tell me if i missed anything

**Comment by Humonitarian** [comment]
I imagine the smart move would be keeping LMG high, but buffing its penetration, just to make armor less effective.

**Comment by MLGTASTICa** [comment]
> > it directly adds wield delay to the AMR and Valkrie
> 
> Good catch, these weapons should have no delay.
> 
> > is single-handedly done to keep LMGS worse-overall in CQC
> 
> Good. LMGs should be hard to deploy, but powerful. That's interesting. Being just bigger mag rifles isn't.
> 
> > to make VIG count more is added
> 
> Vig is already essential for headshots, and provides many other combat advantages.
> 
> > weapon balance can in the current state be done only through Damage , AP , Magsize , Size , Cost and rarity
> 
> Recoil, handedness, etc. are also important. More attention should be given to those qualities. Wield delay is good, but only when it doesn't affect weapons like pistols, or even SMGs. I wouldn't mind it on battlerifles, however, if Valo can be convinced.
> 
> As a final note, I still stand by my opinion of having a hard-limit to how much vigilance can effect it. Instant wield on LMGs is just... Yeah, even when you have 100 vig.

The point is that the wield delay effects shouldnt apply just to LMGS ,because there are weapons far more robust that also need them (wintermule,  vintorez).
Headshots , whatsoever , only provide combat disadvantages , there is NO reason to aim for the head unless the other guy is  not wearing head armor. There is no upside from hitting the head multiple times as long as they have armor for the very simple reason that it will not gib.
Pistols rarely need to get wielded, SMGs require just 10 vigilance to instantly wield and have a very low maximum delay. No weapon should bypass wield timers other than in rare cases(AMR , Valkrie )

**Comment by ValoTheValo** [comment]
What does headshots have to do with this?

**Comment by Humonitarian** [comment]
> > > it directly adds wield delay to the AMR and Valkrie
> > 
> > 
> > Good catch, these weapons should have no delay.
> > > is single-handedly done to keep LMGS worse-overall in CQC
> > 
> > 
> > Good. LMGs should be hard to deploy, but powerful. That's interesting. Being just bigger mag rifles isn't.
> > > to make VIG count more is added
> > 
> > 
> > Vig is already essential for headshots, and provides many other combat advantages.
> > > weapon balance can in the current state be done only through Damage , AP , Magsize , Size , Cost and rarity
> > 
> > 
> > Recoil, handedness, etc. are also important. More attention should be given to those qualities. Wield delay is good, but only when it doesn't affect weapons like pistols, or even SMGs. I wouldn't mind it on battlerifles, however, if Valo can be convinced.
> > As a final note, I still stand by my opinion of having a hard-limit to how much vigilance can effect it. Instant wield on LMGs is just... Yeah, even when you have 100 vig.
> 
> The point is that the wield delay effects shouldnt apply just to LMGS ,because there are weapons far more robust that also need them (wintermule, vintorez). Headshots , whatsoever , only provide combat disadvantages , there is NO reason to aim for the head unless the other guy is not wearing head armor. There is no upside from hitting the head multiple times as long as they have armor for the very simple reason that it will not gib. Pistols rarely need to get wielded, SMGs require just 10 vigilance to instantly wield and have a very low maximum delay. No weapon should bypass wield timers other than in rare cases(AMR , Valkrie )

WYPI (when you play it)

**Comment by Humonitarian** [comment]
PR seems good balance-wise, last change I'd like to see is both LMGs and RPG being moved closer to say, 1.2 wield delay, but the vigilance increased to 1-1.5 so IH can't instant-wield either.

**Comment by ValoTheValo** [comment]
> PR seems good balance-wise, last change I'd like to see is both LMGs and RPG being moved closer to say, 1.2 wield delay, but the vigilance increased to 1-1.5 so IH can't instant-wield either.

1 second and 90 vig sound good?

**Comment by Humonitarian** [comment]
> > PR seems good balance-wise, last change I'd like to see is both LMGs and RPG being moved closer to say, 1.2 wield delay, but the vigilance increased to 1-1.5 so IH can't instant-wield either.
> 
> 1 second and 90 vig sound good?

For now, I suppose yes.

**Comment by TheShown911** [comment]
Neutral on this PR.

**Comment by Humonitarian** [comment]
Don't forget the maxim

**Comment by ValoTheValo** [comment]
> Don't forget the maxim

Ah shit. Dont know if ill be back at my pc in time before merge to do that.

**Comment by ValoTheValo** [comment]
Oh god dammit what does it conflict with aaaaa

**Comment by ValoTheValo** [comment]
i hate merge conflicts.
